File: kotlin/rna-transcription/src/main/kotlin/RnaTranscription.kt
fun transcribeToRna(dna: String): String =
    dna.map { it.toRna() }
        .joinToString("")

internal fun Char.toRna(): Char =
    when (this) {
        'G' -> 'C'
        'C' -> 'G'
        'T' -> 'A'
        'A' -> 'U'
        else -> throw IllegalArgumentException("Unknown nucleotide : $this")
    }

File: dart/rna-transcription/lib/rna_transcription.dart
class RnaTranscription {
  final dnaToRna = {"G": "C", "C": "G", "T": "A", "A": "U"};

  String toRna(String dna) =>
      dna.split("").map((nucleotide) => dnaToRna[nucleotide]).join("");
}
